lang/cs/Org.Apache.REEF.Network/Elastic/Comm/Impl/TopologyUpdate.cs

```diff
@@ -0,0 +1,196 @@
+        /// </summary>
+        /// <param name="node">The node receiving the update</param>
+        /// <param name="children">The update to the children of the node</param>
+        /// <param name="root">The update for the root of the node</param>
+        public TopologyUpdate(string node, List<string> children, string root)
+        {
+            Node = node;
+            Children = children;
+            Root = root;
+        }
+
+        /// <summary>
+        /// Create an update for a node containing only the list of children.
+        /// </summary>
+        /// <param name="node">The node receiving the update</param>
+        /// <param name="children">The update to the children of the node</param>
+        public TopologyUpdate(string node, List<string> children) : this(node, children, string.Empty)
+        {
+        }
+
+        /// <summary>
+        /// Create an update for a node containing only the root node.
+        /// </summary>
+        /// <param name="node">The node receiving the update</param>
+        /// <param name="root">The update for the root of the node</param>
+        public TopologyUpdate(string node, string root) : this(node, new List<string>(), root)
+        {
+        }
+
+        /// <summary>
+        /// The node receiving the update.
+        /// </summary>
+        public string Node { get; private set; }
+
+        /// <summary>
+        /// The updates for the children.
+        /// </summary>
+        public List<string> Children { get; set; }
+
+        /// <summary>
+        /// The updates for the root.
+        /// </summary>
+        public string Root { get; private set; }
+
+        /// <summary>
+        /// The total memory size for the update (used for serialization).
+        /// </summary>
+        public int Size
+        {
+            get
+            {
+                // 1 int for the size of node
+                // The size of node
+                // 1 int for the number of children
+                // 1 int for the length of each children
+                // The size of the string of each child
+                // 1 int + the size of root if not null
+                var nodeSize = sizeof(int) + Node.Length;
+                var childrenSize = sizeof(int) + (Children.Count * sizeof(int)) + Children.Sum(x => x.Length);
+                var rootSize = sizeof(int) + Root.Length;
+
+                return nodeSize + childrenSize + rootSize;
+            }
+        }
+
+        /// <summary>
+        /// Serialize the update.
+        /// </summary>
+        /// <param name="buffer">The memory space where to copy the serialized update</param>
+        /// <param name="offset">Where to start writing in the buffer</param>
+        /// <param name="updates">The updates to serialize</param>
+        internal static void Serialize(byte[] buffer, ref int offset, List<TopologyUpdate> updates)
```

Comment:
By the way, why are we serializing everything like that? is it for Java interoperability?

Reply:
Not really. Where you suggesting to use codecs? I found that the previous group communication uses a similar strategy and I simply piggyback on that. I actually like to have the serialization logic within the message class.